addon/helpers/hook.js

```diff
@@ -4,12 +4,11 @@ import decorateHook from 'ember-hook/utils/decorate-hook';
 import delimit from 'ember-hook/utils/delimit';
 import returnWhenTesting from 'ember-hook/utils/return-when-testing';
 
-const { Helper } = Ember;
+const { Helper, assign } = Ember;
 
-export default Helper.extend({
-  compute(params, qualifiers = {}) {
-    const [hook] = params;
+export function hook([hook, oldQualifiers = {}], newQualifiers = {}) {
+  const qualifiers = assign({}, oldQualifiers, newQualifiers);
+  return returnWhenTesting(config, decorateHook(delimit(hook), qualifiers));
+}
 
-    return returnWhenTesting(config, decorateHook(delimit(hook), qualifiers));
-  }
-});
+export default Helper.helper(hook);
```

Comment:
FYI: the reason I re-formatted this was b/c `ember g` spit out my new helper in this method and it seemed easier to write a test for it if the compute method is exported directly.